rpc/types/src/defaults.rs

```diff
@@ -0,0 +1,78 @@
+//! These functions define the default values
+//! of optional fields in request/response types.
+//!
+//! For example, [`crate::json::GetBlockRequest`]
+//! has a [`crate::json::GetBlockRequest::height`]
+//! field and a [`crate::json::GetBlockRequest::hash`]
+//! field, when the RPC interface reads JSON without
+//! `height`, it will use [`default_height`] to fill that in.
```

Comment:
These functions exist to pass to `serde` and `epee` for default field values.

Our own free functions are used over something like `Default::default` because:
- It's more explicit
- Sometimes we need the non-default, e.g. `default_bool_true`.

There's no example in this PR, but these are extensively used in the other PRs.